server/protocols/jmap/src/main/java/org/apache/james/jmap/model/Message.java

```diff
@@ -222,10 +206,22 @@ public Message build() {
             ImmutableList<Attachment> attachments = this.attachments.build();
             ImmutableMap<BlobId, SubMessage> attachedMessages = this.attachedMessages.build();
             Preconditions.checkState(areAttachedMessagesKeysInAttachments(attachments, attachedMessages), "'attachedMessages' keys must be in 'attachements'");
+            OldKeyword oldKeyword = OldKeyword.fromFlags(flags);
             boolean hasAttachment = hasAttachment(attachments);
-
-            return new Message(id, blobId, threadId, mailboxIds, Optional.ofNullable(inReplyToMessageId), isUnread, isFlagged, isAnswered, isDraft, hasAttachment, headers, Optional.ofNullable(from),
-                    to.build(), cc.build(), bcc.build(), replyTo.build(), subject, date, size, preview, textBody, htmlBody, attachments, attachedMessages);
+            Optional<ImmutableMap<String, Boolean>> mapKeywords = flags.map(flag -> Optional.of(Keywords.factory()
+                    .filterUnsupportedFlags()
+                    .fromFlags(flag)
+                    .asMap()))
+                .orElse(Optional.empty());
+
+            return new Message(id, blobId, threadId, mailboxIds, Optional.ofNullable(inReplyToMessageId),
+                oldKeyword.isUnread().orElse(false),
```

Comment:
You should not pass oldKeyword here.

is* methods should look like this in Message:

```
public boolean isDraft() {
    return keywords.contains(DRAFT);
}
```